routers/web/repo/actions/actions.go

```diff
@@ -132,17 +149,53 @@ func List(ctx *context.Context) {
 		page = 1
 	}
 
-	workflow := ctx.FormString("workflow")
-	actorID := ctx.FormInt64("actor")
-	status := ctx.FormInt("status")
-	ctx.Data["CurWorkflow"] = workflow
-
 	actionsConfig := ctx.Repo.Repository.MustGetUnit(ctx, unit.TypeActions).ActionsConfig()
 	ctx.Data["ActionsConfig"] = actionsConfig
 
-	if len(workflow) > 0 && ctx.Repo.IsAdmin() {
+	if len(workflowID) > 0 && ctx.Repo.IsAdmin() {
 		ctx.Data["AllowDisableOrEnableWorkflow"] = true
-		ctx.Data["CurWorkflowDisabled"] = actionsConfig.IsWorkflowDisabled(workflow)
+		isWorkflowDisabled := actionsConfig.IsWorkflowDisabled(workflowID)
+		ctx.Data["CurWorkflowDisabled"] = isWorkflowDisabled
+
+		if !isWorkflowDisabled && curWorkflow != nil {
+			if events, err := jobparser.ParseRawOn(&curWorkflow.RawOn); err == nil {
+				if err != nil {
+					log.Warn("ignore check invalid workflow events %q: %v", curWorkflow.Name, err)
+				} else {
+					for _, event := range events {
+						if event.Name == webhook_module.HookEventWorkflowDispatch.Event() {
```

Comment:
I don't think this should be defined in webhook.
Just a hard-coding is enough. Maybe we can define a event name list in `gitea/act`, then we can use them without hard-coding string.